ui/src/dashboards/reducers/ui.js

```diff
@@ -109,6 +109,27 @@ export default function ui(state = initialState, action) {
       return {...state, ...newState}
     }
 
+    case 'DELETE_DASHBOARD_CELL': {
+      const {cell} = action.payload
+      const {dashboard} = state
+
+      const index = dashboard.cells.findIndex((c) => c.x === cell.x && c.y === cell.y)
+      const newCells = [
+        ...dashboard.cells.slice(0, index),
```

Comment:
Previously we've used filter for deletion. This works too, and might be more semantic given it's index-based.